drivers/docker/driver_test.go

```diff
@@ -72,7 +72,7 @@ func dockerTask(t *testing.T) (*drivers.TaskConfig, *TaskConfig, []int) {
 	dockerDynamic := ports[1]
 
 	cfg := TaskConfig{
-		Image:     "busybox",
+		Image:     "busybox:latest",
```

Comment:
Since we specify the image to load, what does adding `:latest` do here?

We want to avoid downloading the image on every test, but since we provide the image directly I'm not sure `:latest` will have any effect at all?

Reply:
This change is noop, just clarification, as the tarball image is labeled `:latest`:

```
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker image ls
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ cat ./drivers/docker/test-resources/docker/busybox.tar | docker image load
1834950e52ce: Loading layer [==================================================>]  1.311MB/1.311MB
5f70bf18a086: Loading layer [==================================================>]  1.024kB/1.024kB
Loaded image: busybox:latest
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker image ls
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
busybox             latest              47bcc53f74dc        2 years ago         1.11MB
```

Reply:
I can use the image id instead, e.g. `47bcc53f74dc94b1920f0b34f6036096526296767650f223433fe65c35f149eb`, but that feels not so dev friendly - sadly this image was created prior to 1.10 - so doesn't support content based addressing.  Should I update the image so we can use `busybox@sha256:....`?

Reply:
This change should be a noop and isn't reflection of docker or nomad change in behavior AFAIK.  Rather it simply making an implicit test behavior explicit.  Unless I'm missing something about how Nomad works.

With Docker, no tag implicitly means `:latest` and it does not use any random tag available locally.  The following script demonstrates how docker ignores local image and pulls `:latest`:

```
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker image ls
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ cat ./drivers/docker/test-resources/docker/busybox.tar | docker image load
1834950e52ce: Loading layer [==================================================>]  1.311MB/1.311MB
5f70bf18a086: Loading layer [==================================================>]  1.024kB/1.024kB
Loaded image: busybox:latest
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker run -it --rm busybox echo hi
hi
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker tag busybox:latest busybox:custom
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker image rm busybox:latest
Untagged: busybox:latest
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker image ls
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
busybox             custom              47bcc53f74dc        2 years ago         1.11MB
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ docker run -it --rm busybox echo hi
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
90e01955edcd: Pull complete
Digest: sha256:2a03a6059f21e150ae84b0973863609494aad70f0a80eaeb64bddd8d92465812
Status: Downloaded newer image for busybox:latest
hi
```

Notice how it still pulled a `:latest` image.

Does nomad interpret no tag differently?

Reply:
To avoid any confusion, I dropped this commit from PR - so now it's back to the old implicit style.

Reply:
Oh! I was wrong! No tag was the same as latest. Sorry for the confusion. Explicit is better than implicit. :sweat_smile:

Reply:
No problem, I'll re-add that change to PR and merge then - thanks!